gitpod-io/gitpod
# Hw/ide encrypt

## Description


## Related Issue(s)

Fixes #

## How to test

- Open workspace in preview env with latest browser code
- Check localStorage if `credentials.providers` is encrypted

**compatible#1**
- Close workspace and change default IDE to stable browser code
- Re-open workspace, check if everything is alright (compatible)

**compatible#2 old workspace reopen**
- Copy credentials.provider from production and set
- Re-open workspace use stable browser code
- Check if it encrypted again and the decrypted value is the same with just copied value

## Release Notes

```release-note
NONE
```

## Documentation


## Build Options:

      Experimental feature to run the build with GitHub Actions (and not in Werft).
      leeway-target=components:all
      Run Leeway with `--dont-test`

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
